custom_components/waste_collection_schedule/waste_collection_schedule/service/CitiesAppsCom.py

```diff
@@ -1459,39 +1459,42 @@ def get_specific_citiy(self, search: str) -> dict:
             if city["name"].lower().strip() == search.lower().strip():
                 return city
         raise SourceArgumentNotFoundWithSuggestions(
-            "city", [city["name"] for city in cities]
+            "city", search, [city["name"] for city in cities]
         )
 
-    def get_garbage_calendars(self, city_id: str) -> list:
+    def get_uses_garbage_calendar_v2(self, city_id: str) -> bool:
         r = self._session.get(
-            f"https://api.v2.citiesapps.com/waste-management/by-city/{city_id}/areas")
-        r.raise_for_status()
-        return r.json()["data"]
-
-    def get_specific_calendar(self, city_id: str, search: str) -> dict:
-        calendars = self.get_garbage_calendars(city_id)
-        for calendar in calendars:
-            if calendar["street"].lower().strip() == search.lower().strip():
-                return calendar
-        raise SourceArgumentNotFoundWithSuggestions(
-            "calendar", [calendar["street"] for calendar in calendars]
-        )
-
-    def get_garbage_plans(self, garbage_calendar: dict) -> list:
-
-        r = self._session.get(
-            f"https://api.v2.citiesapps.com/waste-management/areas/{garbage_calendar['_id']}/calendar"
+            f"https://api.citiesapps.com/entities/{city_id}/services"
         )
         r.raise_for_status()
+        essentials = r.json()["essentials"]
 
-        return r.json()["garbageCollectionDays"]
+        return essentials["garbage_calendar_v2"]
 
     def fetch_garbage_plans(self, city: str, calendar: str):
         city_dict = self.get_specific_citiy(city)
-        city_id = city_dict["_id"]
-        specific_calendar = self.get_specific_calendar(city_id, calendar)
+        api = self.GarbageApiV1(self._session)
+        is_v2 = self.get_uses_garbage_calendar_v2(city_dict["_id"])
+
+        if is_v2:
+            api = self.GarbageApiV2(self._session)
 
-        return self.get_garbage_plans(specific_calendar)
+        return {
+            "is_v2": is_v2,
+            "data": api.fetch_garbage_plans(city_dict, calendar)
+        }
+
+    def get_garbage_calendars(self, city_id: str) -> list:
```

Comment:
Type hinted as returning list, but returns `dict[str, bool|list]`

Reply:
Ah, thank you, will update.